lib/scanner.lpp

```diff
@@ -1205,6 +1205,13 @@ RTIME           PT(-?[0-9]+(\.[0-9]+)?[HMSu])+
                                            else {
                                               // re-get the buffer pointer
                                               fn = fname->getBuffer();
+
+                                              // in case fn is in form "ModuleName.SomeModuleClass.qc"
+                                              // convert it to absolute path "/some/implicit/path/ModuleName/SomeModuleClass.qc". see #2966
+                                              QoreString moduleAbsPath(QMM.getModuleAbsPath(fn));
+                                              if (!moduleAbsPath.empty())
+                                                fn = moduleAbsPath.c_str();
```

Comment:
please indent with 4 spaces - some of the old code is indented with 3 spaces, in these cases you can update the indentation of the function or block if you want